test/0.8.9/oracle/hash-consensus-frames.js

```diff
@@ -26,36 +30,109 @@ const computeIthFrameStartTime = (frameIndex, epochsPerFrame, initialEpoch) =>
 const computeNextFrameStartSlot = (time, epochsPerFrame, initialEpoch) =>
   computeIthFrameStartSlot(getFrameIndex(time, epochsPerFrame, initialEpoch), epochsPerFrame, initialEpoch)
 
-
 contract('HashConsensus', ([admin, member1, member2]) => {
-  const INITIAL_EPOCH = 3
+  const TEST_INITIAL_EPOCH = 3
+
+  context('Frame methods', () => {
+    let consensus = null
+    const deploy = async (options = undefined) => {
+      const deployed = await deployHashConsensus(admin, options)
+      consensus = deployed.consensus
+    }
+
+    context('getFrameConfig', () => {
+      before(deploy)
+
+      it('should return initial data', async () => {
+        assert.equal(+(await consensus.getFrameConfig()).epochsPerFrame, EPOCHS_PER_FRAME)
+        assert.equal(+(await consensus.getFrameConfig()).initialEpoch, INITIAL_EPOCH)
+      })
+
+      it('should return new data', async () => {
+        await consensus.setFrameConfig(100, 50)
+
+        assert.equal(+(await consensus.getFrameConfig()).epochsPerFrame, 100)
+        assert.equal(+(await consensus.getFrameConfigFastLaneLengthSlots()), 50)
+        assert.equal(+(await consensus.getFrameConfig()).initialEpoch, INITIAL_EPOCH)
+      })
+    })
+    context('setFrameConfig', () => {
+      beforeEach(deploy)
+
+      it('should set data', async () => {
+        await consensus.setFrameConfig(100, 50)
+
+        assert.equal(+(await consensus.getFrameConfig()).epochsPerFrame, 100)
+        assert.equal(+(await consensus.getFrameConfigFastLaneLengthSlots()), 50)
+        assert.equal(+(await consensus.getFrameConfig()).initialEpoch, INITIAL_EPOCH)
+      })
+
+      it('should set first epoch in next frame', async () => {
+        await consensus.setTimeInEpochs(INITIAL_EPOCH + EPOCHS_PER_FRAME)
+        await consensus.setFrameConfig(100, 50)
+
+        assert.equal(+(await consensus.getFrameConfig()).epochsPerFrame, 100)
+        assert.equal(+(await consensus.getFrameConfigFastLaneLengthSlots()), 50)
+        assert.equal(+(await consensus.getFrameConfig()).initialEpoch, EPOCHS_PER_FRAME + 1)
+      })
+
+      it('should revert if epochsPerFrame == 0', async () => {
+        await assert.revertsWithCustomError(consensus.setFrameConfig(0, 50), 'EpochsPerFrameCannotBeZero()')
+      })
+
+      it('should revert if fastLaneLengthSlots > epochsPerFrame * SLOTS_PER_EPOCH', async () => {
+        await assert.revertsWithCustomError(consensus.setFrameConfig(1, 50), 'FastLanePeriodCannotBeLongerThanFrame()')
+      })
+
+      it('should revert if epoch < config.initialEpoc', async () => {
+        await consensus.setTimeInEpochs(INITIAL_EPOCH - 1)
+
+        await assert.revertsWithCustomError(consensus.setFrameConfig(1, 50), 'InitialEpochIsYetToArrive()')
+      })
+
+      it('should emit FrameConfigSet & FastLaneConfigSet event', async () => {
+        const tx = await consensus.setFrameConfig(100, 50)
+
+        assertEvent(tx, 'FrameConfigSet')
```

Comment:
need to test for event arguments here and later, like it's done e.g. here: https://github.com/lidofinance/lido-dao/blob/feature/shapella-upgrade/test/0.8.9/oracle/hash-consensus-members.test.js#L110

Reply:
Added